src/neo4j_genai/schema.py

```diff
@@ -138,3 +145,54 @@ def get_schema(
             "\n".join(formatted_rels),
         ]
     )
+
+
+def get_structured_schema(driver: neo4j.Driver) -> dict[str, Any]:
+    """
+    Returns the structured schema of the graph.
+
+    Args:
+        driver (neo4j.Driver): Neo4j Python driver instance.
+
+    Returns:
+        dict[str, Any]: the graph schema information in a structured format.
+    """
+    node_properties = [
+        data["output"]
+        for data in query_database(
+            driver,
+            NODE_PROPERTIES_QUERY,
+            params={"EXCLUDED_LABELS": EXCLUDED_LABELS + [BASE_ENTITY_LABEL]},
+        )
+    ]
+
+    rel_properties = [
+        data["output"]
+        for data in query_database(
+            driver, REL_PROPERTIES_QUERY, params={"EXCLUDED_LABELS": EXCLUDED_RELS}
+        )
+    ]
+
+    relationships = [
+        data["output"]
+        for data in query_database(
+            driver,
+            REL_QUERY,
+            params={"EXCLUDED_LABELS": EXCLUDED_LABELS + [BASE_ENTITY_LABEL]},
+        )
+    ]
+
+    # Get constraints and indexes
+    try:
+        constraint = query_database(driver, "SHOW CONSTRAINTS")
+        index = query_database(driver, INDEX_QUERY)
+    except ClientError:
```

Comment:
In which case would these queries fail? Apoc missing?

Reply:
Not all users are allowed to create constraint and indexes. Apparently it's not allowed at all in Community edition, because the E2E tests are failing because of it, so I will have to setup a different test depending on Neo4j edition 😅

Reply:
Sorry, we're not creating anything here (I was thinking about the tests). The problem here is read-only user. I can add a comment.